### PR TITLE
Husky ur5 control integration (adding the husky ur5 arm)

### PR DESCRIPTION
**Pull request by UltronDestroyer**

### CODE REVIEW

File: config/control_with_ur5_arm.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+husky_joint_publisher:
+  type: "joint_state_controller/JointStateController"
```
Comment (paulbovbel) [inline config/control_with_ur5_arm.yaml]: unpleasant to repeat all the stuff from control.yaml. Can you make a separate control_ur5.yaml?

File: launch/control.launch [hunk 2]
```diff
@@ -1,7 +1,29 @@
 <?xml version="1.0"?>
 <launch>
-  <rosparam command="load" file="$(find husky_control)/config/control.yaml" />
-  <node name="controller_spawner" pkg="controller_manager" type="spawner" args="husky_joint_publisher husky_velocity_controller"/>
+
+  <arg name="ur5_enabled" default="false"/>
+
```
Comment (paulbovbel) [inline launch/control.launch:5]: rosparam load control.yaml
  <node name="controller_spawner" pkg="controller_manager" type="spawner" args="husky_joint_publisher husky_velocity_controller"/>

File: launch/control.launch [hunk 3]
```diff
@@ -1,7 +1,29 @@
...
+  <!-- If we're using the ur5 arm -->
+  <group if="$(arg ur5_enabled)">
+       <!-- <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher"/> -->
+
+       <!-- Load UR5 controllers -->
+       <rosparam command="load" file="$(find husky_control)/config/control_with_ur5_arm.yaml" />
```
Comment (paulbovbel) [inline launch/control.launch]: rosparam load control_ur5.yaml

Pretty sure you can just run a separate spawner instance for arm_controller no? Maybe rename to husky_arm_controller for consistency as well.

File: launch/control.launch [hunk 4]
```diff
@@ -1,7 +1,29 @@
...
+       <!-- Fake Calibration -->
+  <node pkg="rostopic" type="rostopic" name="fake_joint_calibration"
+        args="pub /calibrated std_msgs/Bool true" />
+
+       <!-- Diagnostics -->
+       <node pkg="pr2_mechanism_diagnostics" type="pr2_mechanism_diagnostics" name="pr2_mechanism_diagnostics" />
+  </group>
+
+  <!-- If not using UR5 arm -->
+  <group unless="$(arg ur5_enabled)">
```
Comment (paulbovbel) [inline launch/control.launch]: Really don't need this group

File: launch/control.launch (hunk 2 above)
Comment (UltronDestroyer) [inline launch/control.launch:5]: ok, done

File: launch/control.launch (hunk 3 above)
Comment (UltronDestroyer) [inline launch/control.launch]: ok, done

File: config/control_with_ur5_arm.yaml (hunk 1 above)
Comment (UltronDestroyer) [inline config/control_with_ur5_arm.yaml]: ok, done.

File: launch/control.launch (hunk 4 above)
Comment (UltronDestroyer) [inline launch/control.launch]: which group

Comment (UltronDestroyer) [inline launch/control.launch]: without the group then the controller_manager will attempt to spawn the controllers again.

this solution works, sure there are others, but this code isn't redundant at all.

File: config/control_ur5.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+arm_controller:
```
Comment (paulbovbel) [inline config/control_ur5.yaml:1]: can we rename this husky_arm_controller or something for consistency?

Comment (UltronDestroyer) [inline config/control_ur5.yaml:1]: I tried doing that, but it broke a bunch of other things. Given the current timeline we have.. I don't have time to poke around and I'd rather get it working on the real robot before making enhancements, but feel free to change it if you have time.

Comment (paulbovbel) [inline config/control_ur5.yaml:1]: Breaks what things? Should be a pretty straightforward change
`find . -type f -exec sed -i 's/arm_controller/husky_arm_controller/g' {} +`

Comment (UltronDestroyer) [inline config/control_ur5.yaml:1]: MoveIt becomes unresponsive. I'm not sure what else had to be changed, but it's not as simple as changing arm_controller -> husky_arm_controller.

**Comment by paulbovbel** [comment]
LGTM

**Pull request merged**